modules/imgproc/test/test_houghCircles.cpp

```diff
@@ -0,0 +1,233 @@
+        Mat src = imread(filename, IMREAD_GRAYSCALE);
+        EXPECT_FALSE(src.empty()) << "Invalid test image: " << filename;
+
+        GaussianBlur(src, src, Size(9, 9), 2, 2);
+
+        vector<Vec3f> circles;
+        const double dp = 1.0;
+        HoughCircles(src, circles, CV_HOUGH_GRADIENT, dp, minDist, edgeThreshold, accumThreshold, minRadius, maxRadius);
+
+        string imgProc = string(cvtest::TS::ptr()->get_data_path()) + "imgproc/";
+#if DEBUG_IMAGES
+        highlightCircles(filename, circles, imgProc + test_case_name + ".png");
+#endif
+
+        string xml = imgProc + "HoughCircles.xml";
+        FileStorage fs(xml, FileStorage::READ);
+        FileNode node = fs[test_case_name];
+        if (node.empty())
+        {
+            fs.release();
+            fs.open(xml, FileStorage::APPEND);
+            EXPECT_TRUE(fs.isOpened()) << "Cannot open sanity data file: " << xml;
+            fs << test_case_name << circles;
+            fs.release();
+            fs.open(xml, FileStorage::READ);
+            EXPECT_TRUE(fs.isOpened()) << "Cannot open sanity data file: " << xml;
+        }
+
+        vector<Vec3f> exp_circles;
+        read(fs[test_case_name], exp_circles, vector<Vec3f>());
+        fs.release();
+
+        EXPECT_EQ(exp_circles.size(), circles.size());
+    }
+};
+
+TEST_P(HoughCirclesTestFixture, regression)
+{
+    run_test();
+}
+
+INSTANTIATE_TEST_CASE_P(ImgProc, HoughCirclesTestFixture, testing::Combine(
+    // picture_name:
+    testing::Values("imgproc/stuff.jpg"),
+    // minDist:
+    testing::Values(20),
+    // edgeThreshold:
+    testing::Values(20),
+    // accumThreshold:
+    testing::Values(30),
+    // minRadius:
+    testing::Values(20),
+    // maxRadius:
+    testing::Values(200)
+    ));
+
+TEST(HoughCirclesTest, DefaultMaxRadius)
+{
+    string picture_name = "imgproc/stuff.jpg";
+    const double dp = 1.0;
+    double minDist = 20;
+    double edgeThreshold = 20;
+    double accumThreshold = 30;
+    int minRadius = 20;
+    int maxRadius = 0;
+
+    string filename = cvtest::TS::ptr()->get_data_path() + picture_name;
+    Mat src = imread(filename, IMREAD_GRAYSCALE);
+    EXPECT_FALSE(src.empty()) << "Invalid test image: " << filename;
+
+    GaussianBlur(src, src, Size(9, 9), 2, 2);
+
+    vector<Vec3f> circles;
+    HoughCircles(src, circles, CV_HOUGH_GRADIENT, dp, minDist, edgeThreshold, accumThreshold, minRadius, maxRadius);
+
+#if DEBUG_IMAGES
+    string imgProc = string(cvtest::TS::ptr()->get_data_path()) + "imgproc/";
+    highlightCircles(filename, circles, imgProc + "HoughCirclesTest_DefaultMaxRadius.png");
+#endif
+
+    int maxDimension = std::max(src.rows, src.cols);
+
+    EXPECT_GT(circles.size(), size_t(0)) << "Should find at least some circles";
+    for (size_t i = 0; i < circles.size(); ++i)
+    {
+        EXPECT_GE(circles[0][2], minRadius) << "Radius should be >= minRadius";
+        EXPECT_LE(circles[0][2], maxDimension) << "Radius should be <= max image dimension";
```

Comment:
`i` index seems to be forgotten here

Reply:
Thanks for catching that. I just pushed a fix.